addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -307,7 +307,14 @@ const Wysiwyg = Widget.extend({
                 $el.selectElement();
 
                 if (!$el.parent().hasClass('o_stars')) {
-                    self.openMediaDialog(params);
+                    // Waiting for all the options to be initialized before
+                    // opening the media dialog and only if the media has not
+                    // been deleted in the meantime.
+                    self.snippetsMenu._mutex.getUnlockedDef().then(() => {
+                        if ($el[0].parentElement) {
+                            self.openMediaDialog(params);
```

Comment:
The explanation seems good but I don't think you can do it like this:
- I don't think `snippetsMenu` is necessarily defined here (I just tested: Projects -> Open a task -> add an image -> double click on it -> crash)
- Accessing `_mutex` like that should be avoided (private member of the class).

Maybe do this:
- Create a method in this class, something like `onceUnlocked()` -> by default a resolved promise
- Where we define `snippetsMenu` (if not here), implements that method to listen to the snippetsMenu unlocked stuff
- Create a public method in snippetsMenu to access that

Note maybe `_mutex.getUnlockedDef()` is not the best way here, I think you would rather want to wait for an empty action in the mutex stack. Something like `_mutex.exec(() => null)`.
- Click on image -> add option creation to mutex queue
- Double click on image
- Click on a snippet -> add snippet activation to mutex queue
=> With what I suggest media dialog opens once options are ready, with yours, I think it will open after the snippet that was clicked after was activated.

Of course, as your commit explains, it's theorical and probably only a tour problem. In practice, would be difficult to click quick enough to have the issue :wink: